### ISSUE TITLE
[Feature]: Nesterov Accelerated Gradient (NAG-GD) Sampler

### ISSUE THREAD

**Issue by soran-ghaderi** [soran-ghaderi/torchebm#81]
### Prerequisites

- [x] I have searched existing issues and discussions to ensure this feature hasn't been requested before
- [x] I have read the documentation and confirmed this feature doesn't already exist

### Problem Summary

Lacks Nesterov Accelerated Gradient (NAG-GD) Sampler

### Proposed Solution

It applies a look-ahead step at each update and evaluates the gradient at that look-ahead point.

NAG for strongly convex functions as a high-resolution ODE, adds noise, and discretizes for sampling.

Introduces a hyperparameter to control acceleration, distinct from time-rescaling.

Can mix faster than Langevin dynamics (in the continuous and discrete cases) for certain distributions, with empirical evidence supporting the theoretical results, including Bayesian neural networks and multi-modal targets.​

Neither purely overdamped nor underdamped Langevin, blending optimization acceleration with stochastic sampling.

### Feature Category

Samplers (HMC, Langevin, etc.)

### Priority

High - Important for my use case

### Example Usage

```python
from torchebm.samplers.langevin_dynamics import NesterovAcceleratedGradient

nestrov_ag_sampler = NesterovAcceleratedGradient(
    energy_function=energy_fn, step_size=5e-3, device=device
)

final_samples = nestrov_ag_sampler.sample(
    dim=10, n_steps=500, n_samples=10000, return_trajectory=False
)
```

### Implementation

- [ ] I would be willing to contribute to the implementation
- [ ] I can help with testing the feature
- [ ] I can help with documentation
- [ ] I can provide additional use cases/examples

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
